### PR TITLE
Adding `bulk_core_count` to `HostConfiguration`

### PR DESCRIPTION
**Pull request by BradleyOlson64**
Still a lot of work to be done, creating an appropriate storage migration and creating a call to populate the new `bulk_core_count` field.

I made a couple assumptions so far:
1. `bulk_core_count` is cached by `session_index`, since it should only change at session boundaries.
2. The storage and runtime api versions are changed by the addition of `bulk_core_count`, to 11 and 10 respectively.

TODO Reminder: 
- Rename `on-demand-core-count` to `bulk_core_count`. Remove `on-demand-core-count` logic.
- Move `bulk_core_count` to runtime api version 9 if possible and include it in vstaging

### CODE REVIEW

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable 3/3
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/4551296

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/4551279

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable 1/3
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/4551294

File: polkadot/node/core/runtime-api/src/lib.rs [hunk 1]
```diff
@@ -612,5 +622,12 @@ where
 			sender,
 			result = (index)
 		),
+		Request::BulkCoreCount(index, sender) => query!(
```
Comment (eskimor) [inline polkadot/node/core/runtime-api/src/lib.rs:625]: Why do we need a runtime API for this? The node should not care at all about the number of bulk cores.

File: polkadot/runtime/parachains/src/configuration.rs [hunk 2]
```diff
@@ -263,6 +263,8 @@ pub struct HostConfiguration<BlockNumber> {
 	pub minimum_backing_votes: u32,
 	/// Node features enablement.
 	pub node_features: NodeFeatures,
+	/// Bulk core count
+	pub bulk_core_count: u16,
```
Comment (eskimor) [inline polkadot/runtime/parachains/src/configuration.rs:267]: As mentioned on the ticket. I would suggest to re-use/rename the on-demand-core-count configuration, as it is replaced by this one.

File: polkadot/node/core/runtime-api/src/lib.rs (hunk 1 above)
Comment (BradleyOlson64) [reply polkadot/node/core/runtime-api/src/lib.rs:625]: Gotcha, that makes sense.

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/4563719

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/4575243

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable 2/3
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/4575259

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable 1/3
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/4575258

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable 2/3
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/4575840

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable 3/3
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/4575841

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/4576594

**Comment by eskimor** [comment]
Thanks Bradley! Took the necessary pieces.

**Pull request closed without merging**